plugins/modules/rpm_repository.py

```diff
@@ -137,10 +137,11 @@ def main():
             remote.find(failsafe=False)
             desired_attributes["remote"] = remote.href
 
-        repo_config = module.params["repo_config"]
         # Encode the repo_config unless its a string, then assume it is pre-formatted JSON
-        if repo_config is not None and not isinstance(repo_config, string_types):
-            desired_attributes["repo_config"] = json.dumps(repo_config)
+        if module.params["repo_config"] is not None and isinstance(
+            module.params["repo_config"], string_types
+        ):
```

Comment:
Let's try to stay in the domain of the problem here too (We should be done with `module.params` at this point):
```suggestion
        if "repo_config" in desired_attributes and isinstance(
            desired_attributes["repo_config"], string_types
        ):
```

Reply:
Sure. I've made this change as requested